### PR TITLE
Add new remote policy for converting between xml and json content

### PR DESCRIPTION
**Pull request by morganlaw**

### CODE REVIEW

File: remote-policy-xml-json-conversion/pom.xml [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+      <version>1.7.30</version>
+    </dependency>
+    <dependency>
+      <groupId>org.json</groupId>
+      <artifactId>json</artifactId>
+      <version>20200518</version>
+    </dependency>
+  </dependencies>
+
+  <!-- Required for Java 11 functions in the inline editor -->
```
Comment (mju) [inline remote-policy-xml-json-conversion/pom.xml:66]: java 11?

File: remote-policy-xml-json-conversion/src/main/java/com/google/apigee/XmlJsonConversion.java [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+  /**
+   * Converts JSON content to XML.
+   *
+   * @param execution Execute.Execution object to write to.
+   * @param content   String content to convert.
+   * @return Execute.Execution object with converted content.
+   * @throws Exception
+   */
+  private Execute.Execution jsonToXml(Execute.Execution execution, String content)
+      throws Exception {
```
Comment (mju) [inline remote-policy-xml-json-conversion/src/main/java/com/google/apigee/XmlJsonConversion.java:112]: i'd be more specific here.

File: remote-policy-xml-json-conversion/pom.xml (hunk 1 above)
Comment (morganlaw) [reply remote-policy-xml-json-conversion/pom.xml:66]: Cloud functions uses Java 11, the pom.xml is modified from the one they give

File: remote-policy-xml-json-conversion/src/main/java/com/google/apigee/XmlJsonConversion.java [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+   * @throws Exception
+   */
+  private Execute.Execution convert(Execute.Execution execution)
+      throws UnsupportedEncodingException {
+    validateExecution(execution);
+    String conversion = execution.getMessageContext().getTargetRequestMessage()
+        .getFlowVariablesMap().get(CONVERSION_FLOW_VARIABLE).getFlowVariable();
+    String content = execution.getMessageContext().getTargetRequestMessage().getContent()
+        .toString("UTF-8");
+    if (XML_TO_JSON.equals(conversion)) {
```
Comment (mju) [inline remote-policy-xml-json-conversion/src/main/java/com/google/apigee/XmlJsonConversion.java:80]: what happens if both branches evaluate to false?  Do we need some special treatment for that case?

Comment (morganlaw) [reply remote-policy-xml-json-conversion/src/main/java/com/google/apigee/XmlJsonConversion.java:80]: Added an exception thrown for that case. Before I just had it return the same Execution message without modification, but an exception makes more sense in my opinion.

File: remote-policy-xml-json-conversion/src/main/java/com/google/apigee/XmlJsonConversion.java [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+    String conversion = execution.getMessageContext().getTargetRequestMessage()
+        .getFlowVariablesMap().get(CONVERSION_FLOW_VARIABLE).getFlowVariable();
+    String content = execution.getMessageContext().getTargetRequestMessage().getContent()
+        .toStringUtf8();
+    if (XML_TO_JSON.equals(conversion)) {
+      execution = xmlToJson(execution, content);
+    } else if (JSON_TO_XML.equals(conversion)) {
+      execution = jsonToXml(execution, content);
+    } else {
+      throw new IllegalArgumentException("invalid conversion variable");
```
Comment (mju) [inline remote-policy-xml-json-conversion/src/main/java/com/google/apigee/XmlJsonConversion.java:85]: print the what's in conversion?  That way debugging is easier on the server side.

File: remote-policy-xml-json-conversion/src/test/java/com/google/apigee/XmlJsonConversionTest.java [hunk 5]
```diff
@@ -0,0 +1,308 @@
...
+import org.mockito.MockitoAnnotations;
+
+public class XmlJsonConversionTest {
+
+  private static final String CONTENT_JSON = "{\\\"sample\\\":{\\\"heading\\\":\\\"heading\\\",\\\"body\\\":\\\"message\\\"}}";
+  private static final String CONTENT_XML = "<sample><heading>heading</heading><body>message</body></sample>";
+  private static final String CONVERSION_KEY = "conversion";
+  private static final String CONVERSION_VALUE_JSON_TO_XML = "jsontoxml";
+  private static final String CONVERSION_VALUE_XML_TO_JSON = "xmltojson";
+  private XmlJsonConversion XMLJsonConversion;
```
Comment (mju) [inline remote-policy-xml-json-conversion/src/test/java/com/google/apigee/XmlJsonConversionTest.java:26]: xmlJsonConversion

**Pull request merged**